doc/source/release-notes/Percona-XtraDB-Cluster-5.7.20-29.24.rst

```diff
@@ -0,0 +1,121 @@
+  mode, which could lead to data inconsistency. The node will now abort in
+  this case. Bug fixed :bug:`1728774`
+
+* :jirabug:`PXC-806`: fixed early read of the query_id, ensuring valid
+  id to be assigned to the running thread.
+
+* :jirabug:`PXC-850`: ensured that a node, because of data inconsistency,
+  isolates itself before leaving the cluster, thus allowing pending nodes
+  to re-evaluate the quorum. Bug fixed :bug:`1704404`
+
+* :jirabug:`PXC-867`: ``wsrep_sst_rsync`` script was overwriting wsrep debug
+  configuration setting making it not to be taken into account.
+
+* Fixed formatting issue in the error message appearing
+  when SST is not possible due to a timeout. Bug fixed :bug:`1720094`
+
+* :jirabug:`PXC-874`: PXC acting as async slave reported unhandled transaction
+  errors, namely "Rolling back unfinished transaction".
+
+* :jirabug:`PXC-875`: fixed an issue where toggling :variable:`wsrep_provider`
+  off and on failed to reset some internal variables and resulted in PXC
+  logging an "Unsupported protocol downgrade" warning. Bug fixed :bug:`1379204`
+
+* :jirabug:`PXC-877`: fixed PXC hang caused by an internal deadlock.
+
+* Thread failed to mark exit from the InnoDB server concurrency and therefore
+  never got un-register in InnoDB concurrency system.
+
+* :jirabug:`PXC-879`: wrong count took place when ``LOAD DATA`` commands used
+  with GTIDs were executed on one node making other nodes to receive less rows
+  than the first one. Bug fixed :bug:`1741818`
+
+* :jirabug:`PXC-880`: insert to table without primary key was possible with
+  insertable view if :variable:`pxc_strict_mode` variable was set to ENFORCING.
+  Bug fixed :bug:`1722493`
+
+* :jirabug:`PXC-883`: fixed ``ROLLBACK TO SAVEPOINT`` incorrect operation on
+  slaves by avoiding useless wsrep plugin register for a savepoint rollback.
+  Bug fixed :bug:`1700593`
+
+* Fixed IST hang when ``keyring_file_data`` is set. Bug fixed :bug:`1728688`
+
+* :jirabug:`PXC-887`: gcache page files were unnecessarily created due to
+  an error in projecting gcache free size when configured to recover on
+  restart.
+
+* Fixed transaction loss after recovery by avoiding
+  interruption of the binlog recovery based on wsrep saved position.
+  Bug fixed :bug:1734113
+
+* :jirabug:`PXC-897`: fixed empty :variable:`gtid_executed` variable after
+  recovering the position of a node with ``--wsrep_recover``.
+
+* Fixed certification failure in the case of a node
+  restarting at the same time when frequent ``TRUNCATE TABLE`` commands and
+  DML writes occur simultaneously on other nodes. Bug fixed :bug:`1737731`
+
+* :jirabug:`PXC-909`: qpress package was turned into a dependency from
+  suggested/recommended one on Debian 9.
+
+* :jirabug:`PXC-903` and :jirabug:`PXC-910`: init.d/systemctl scripts on Debian
+  9 were updated to avoid starting wsrep-recover if there was no crash, and to
+  fix infinite loop at mysqladmin ping fail because of nonexistent ping user.
```

Comment:
and to fix an infinite loop that occurred due to mysqladmin ping failing because of a nonexistent ping user.